### PR TITLE
Update DocumentRepository.php

### LINKED ISSUE #681: [QueryBuilder] findBy with IN
The query builder should support passing an array as values and do an `IN` query or use `orX`. See #680.

Documentation: The feature was already documented even though its not implemented: https://github.com/doctrine/phpcr-odm-documentation/pull/79

### PR DESCRIPTION
**Pull request by vudaltsov**
Added `$repository->findBy(array('field' => array('val1', 'val2')))` support via `->orX()` method

fix #681

### CODE REVIEW

File: lib/Doctrine/ODM/PHPCR/DocumentRepository.php [hunk 1]
```diff
@@ -168,7 +168,14 @@ public function findBy(array $criteria, array $orderBy = null, $limit = null, $o
                 $where = $qb->andWhere();
             }
 
-            $this->constraintField($where, $field, $value, 'a');
+            if (!is_array($value)) {
```
Comment (dbu) [inline lib/Doctrine/ODM/PHPCR/DocumentRepository.php]: for code readability, i would prefer to switch the cases, and have `if (is_array($value))`.

**Comment by dbu** [comment]
very cool, thanks!

**Comment by lsmith77** [comment]
style ci issues should be fixed via https://github.com/doctrine/phpcr-odm/pull/685

**Comment by vudaltsov** [comment]
@lsmith77, I probably lack some knowledge. How can I do that?

**Comment by lsmith77** [comment]
@vudaltsov no worries .. 

@dbu is #685 ready to merge? in that case @vudaltsov could then rebase on master .. or shall we just merge this PR knowing that the style CI issues are not related to this PR?

**Comment by dbu** [comment]
don't worry about the styleci fail, i will try to get the other PR about that wrapped up.

@vudaltsov can you please swap the order in the `if` statement? see my comment above. once that is done, we can merge this. oh, please also add a note about this fix to the CHANGELOG.md file (add a new line above the version number)

**Comment by vudaltsov** [comment]
@dbu, @lsmith77, hope, I've done everything correctly

**Comment by vudaltsov** [comment]
@dbu smth somehow happened to the CHANGELOG.md on the last line - probably the github online editor added a line ending symbol or smth. The contents didn't change over there

**Comment by dbu** [comment]
thanks a lot!
funny with that last line, usually github displays a "no newline at end of file" in those cases. but i could not see any change, so whatever.

**Pull request merged**